java/choose-camera-controller/src/main/java/com/esri/arcgisruntime/sample/choosecameracontroller/MainActivity.java

```diff
@@ -0,0 +1,195 @@
+
+  private static final String TAG = MainActivity.class.getSimpleName();
+
+  private SceneView mSceneView;
+  private Graphic mPlane3D;
+  private GraphicsOverlay mSceneOverlay;
+  private OrbitGeoElementCameraController mOrbitAeroplaneCameraController;
+  private OrbitLocationCameraController mOrbitLocationCameraController;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    // load plane model from assets into cache directory
+    copyFileFromAssetsToCache(getString(R.string.bristol_model));
+    copyFileFromAssetsToCache(getString(R.string.bristol_skin));
+
+    setupToolbar();
+
+    // create a scene and add it to the scene view
+    mSceneView = findViewById(R.id.sceneView);
+    ArcGISScene scene = new ArcGISScene(Basemap.createImagery());
+    mSceneView.setScene(scene);
+
+    // add base surface for elevation data
+    Surface surface = new Surface();
+    ArcGISTiledElevationSource elevationSource = new ArcGISTiledElevationSource(
+        getString(R.string.world_elevation_service_url));
+    surface.getElevationSources().add(elevationSource);
+    scene.setBaseSurface(surface);
+
+    // create a graphics overlay for the scene
+    mSceneOverlay = new GraphicsOverlay();
+    mSceneOverlay.getSceneProperties().setSurfacePlacement(LayerSceneProperties.SurfacePlacement.ABSOLUTE);
+    mSceneView.getGraphicsOverlays().add(mSceneOverlay);
+
+    // create a camera and set it as the viewpoint for when the scene loads
+    Camera camera = new Camera(38.459291, -109.937576, 5500, 150.0, 20.0, 0.0);
+    mSceneView.setViewpointCamera(camera);
+
+    // instantiate a new camera controller which orbits a target location
+    Point locationPoint = new Point(-109.929589, 38.437304, 1700, SpatialReferences.getWgs84());
+    mOrbitLocationCameraController = new OrbitLocationCameraController(locationPoint, 5000);
+    mOrbitLocationCameraController.setCameraPitchOffset(3);
+    mOrbitLocationCameraController.setCameraHeadingOffset(150);
+
+    loadModel().addDoneLoadingListener(() -> {
+      // instantiate a new camera controller which orbits the aeroplane at a set distance
+      mOrbitAeroplaneCameraController = new OrbitGeoElementCameraController(mPlane3D,
+          100.0);
+      mOrbitAeroplaneCameraController.setCameraPitchOffset(30);
+      mOrbitAeroplaneCameraController.setCameraHeadingOffset(150);
+    });
+  }
+
+  private void setupToolbar() {
+    Toolbar toolbar = findViewById(R.id.toolbar);
+    setSupportActionBar(toolbar);
+    toolbar.setOverflowIcon(ContextCompat.getDrawable(this, R.drawable.ic_photo_camera));
+  }
+
+  @Override public boolean onCreateOptionsMenu(Menu menu) {
+    getMenuInflater().inflate(R.menu.camera_controller_menu, menu);
+    return super.onCreateOptionsMenu(menu);
+  }
+
+  @Override public boolean onOptionsItemSelected(MenuItem item) {
+    switch (item.getItemId()) {
```

Comment:
This needs to be an if statement, only because when these app modules are converted to library modules in the sample viewer it throws this error:

https://stackoverflow.com/questions/51401999/how-to-solve-error-resource-ids-cannot-be-used-in-switch-statement-in-android/51402146